streams/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java

```diff
@@ -291,21 +279,18 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
     private static class FutureSubscriptionInfo {
         private final int version;
         private final UUID processId;
-        private final Set<TaskId> prevTasks;
-        private final Set<TaskId> standbyTasks;
         private final String userEndPoint;
+        private final Map<TaskId, Integer> taskLags;
 
         // for testing only; don't apply version checks
         FutureSubscriptionInfo(final int version,
                                final UUID processId,
-                               final Set<TaskId> prevTasks,
-                               final Set<TaskId> standbyTasks,
-                               final String userEndPoint) {
+                               final String userEndPoint,
+                               final Map<TaskId, Integer> taskLags) {
```

Comment:
I'm not sure this should be necessary either. IIUC, the "future" subscription info isn't supposed to really be a descendant of the current protocol, just a stand-in for _some_ protocol version bigger than ours, in which case all that really matters is the version number. Its roll is just to join the cluster and get downgraded to the "latest" version, in which case it should be able to defer to SubscriptionInfo.

Reply:
Good point